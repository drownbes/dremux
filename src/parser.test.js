const { parse, replace } = require('./parser');


const tests = [
  `
    "<| somefile/name/1.js |>"
    "<| somefile/name/2.js |>"
    "<| somefile/name/3.js |>"
  `,
  `
    "<| somefile/name/1.js |>" "<| somefile/name/2.js |>"
    "<| somefile/name/3.js |>"
  `,
  `
    "<| somefile/name/1.js |>""<| somefile/name/2.js |>"
    "<| somefile/name/3.js |>"
  `,
  ``,
  `"<|`,
  `|>"`
];

const replaceTestData = `{
  "tourOperator":  "APO",
  "bookingClass": "Y",
  "flights": [
    "<| A3293.flight.json |>",
    "<| A3294.flight.json |>"
  ],
  "passengers": "<| TwoAdults.passenger.json |>",
  "baseUrl": "shop/aegean",
  "shopName": "Aegean Airlines"
}`;

const replaceHash = {
  'A3293.flight.json': '1',
  'A3294.flight.json': '2',
  'TwoAdults.passenger.json': '3'
};

const expected = `{
  "tourOperator":  "APO",
  "bookingClass": "Y",
  "flights": [
    1,
    2
  ],
  "passengers": 3,
  "baseUrl": "shop/aegean",
  "shopName": "Aegean Airlines"
}`;


test('parse', () => {
  const p = parse();
  tests.forEach(test => {
    expect(p(test)).toMatchSnapshot();
  });
});

test('replace', () => {
  const p = parse();
  const matches = p(replaceTestData);

  const r = replace(replaceHash)(matches)(replaceTestData);
  expect(r).toEqual(expected);
});

test('replace throw on mis in replace hash', () => {
  const p = parse();
  const matches = p(replaceTestData);

  expect(() => replace({})(matches)(replaceTestData)).toThrow();
});

test('replace return passed string on empty string or empty matches', () => {
  expect(replace({})([])(replaceTestData)).toEqual(replaceTestData);
  expect(replace({})([])('')).toEqual('');
});