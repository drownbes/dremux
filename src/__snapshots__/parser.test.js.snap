// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse 1`] = `
Array [
  Object {
    "end": 31,
    "match": "somefile/name/1.js",
    "start": 5,
  },
  Object {
    "end": 62,
    "match": "somefile/name/2.js",
    "start": 36,
  },
  Object {
    "end": 93,
    "match": "somefile/name/3.js",
    "start": 67,
  },
]
`;

exports[`parse 2`] = `
Array [
  Object {
    "end": 31,
    "match": "somefile/name/1.js",
    "start": 5,
  },
  Object {
    "end": 58,
    "match": "somefile/name/2.js",
    "start": 32,
  },
  Object {
    "end": 89,
    "match": "somefile/name/3.js",
    "start": 63,
  },
]
`;

exports[`parse 3`] = `
Array [
  Object {
    "end": 31,
    "match": "somefile/name/1.js",
    "start": 5,
  },
  Object {
    "end": 57,
    "match": "somefile/name/2.js",
    "start": 31,
  },
  Object {
    "end": 88,
    "match": "somefile/name/3.js",
    "start": 62,
  },
]
`;

exports[`parse 4`] = `Array []`;

exports[`parse 5`] = `Array []`;

exports[`parse 6`] = `Array []`;
